students/john_navitsky/session06/mailroom.py

```diff
@@ -200,62 +253,51 @@ def thank_you_entry():
 
         # check for a list directive
         if selection.lower() in ["l", "list"]:
-            print_report()
+            print_report(donors)
+            continue
+
+        # protect against no entry
+        if not selection:
             continue
 
         # parse the string into name components
         current_donor=parse_name(selection)
 
-        # determine if the provided name matches an existing record
-        donor_id=None
-        for donor_index, existing_donor in enumerate(donors):
-            # assemble full name from first_name, last_name
-            existing_full_name = existing_donor["first_name"]+" "+existing_donor["last_name"]
-            # if we get a match, store the index of the donor record
-            if existing_full_name.lower().strip() == current_donor["full_name"].lower().strip():
-                donor_id=donor_index
+        # reject blatantly bad input
+        if current_donor["first_name"] == "" or current_donor["last_name"] == "":
```

Comment:
since you are parsing the name elsewhere, best to put the validation and parsing all in one place.